src/components/buttons/icon-button.tsx

```diff
@@ -0,0 +1,35 @@
+import { lighten } from 'polished';
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
+  border: none;
+  border-radius: 30px;
+  cursor: pointer;
+
+  height: 24px;
+  width: 24px;
+
+  background: linear-gradient(275.41deg, #543fd7 0%, #2756fd 100%);
+
+  color: #ffffff;
+
+  padding: 0;
+
+  &:hover {
+    background: #3e61ee;
+  }
+
+  &:active {
+    background: ${lighten(0.1, '#3e61ee')};
+  }
+`;
+
+interface CopyProps {
+  onClick: React.MouseEventHandler;
```

Comment:
I would not expose the `onClick` instead see  this general issue:
https://github.com/satellytes/satellytes.com/issues/228

Passing in `onClick` feels super weird for things that support that by default. It's always a big warning sign so please be careful with that in the future.

Reply:
To make it more clear: The idea is to use `export interface CopyProps extends React.ComponentPropsWithoutRef<"button">` and then to spread the `props` (by the way, they should really be called `IconButtonProps` instead of `CopyProps`).

I don't think of this as such a big deal. The component receives an `onClick` and passes it on to the native `button` element. What's wrong with that approach? It's very explicit but other then that I dont't see any flaws. To the outside (users of this component) it doesn't make a difference, does it?

Reply:
Yes it works, but it's not idiomatic to React. That's the deal. I could ask instead why not use `type CopyProps = any`, spread the props and say that people don't need the types because buttons always have `onClick`. Still works and has no flaws.

Instead I went the route following the framework's rails and suggested to use the given button type that helps us to map react components back to the web standard so you can use the `<IconButton/>` as you would use any other `<button/>`. This resonates so much more with me then any custom mapping of `onClick` which creates more noise then the special type `ComponentPropsWithoutRef` I referenced before.

Reply:
Thank you for your suggestions, I do like @georgiee idea and changed it, just like he suggested it. I hope that is fine for everyone.